### PR TITLE
`MIGRATING.md` update / examples for type safe bounds

### PR DESCRIPTION
**Pull request by maurolacy**
Adds entry for migrating to type-safe bounds.

### CODE REVIEW

File: MIGRATING.md [hunk 1]
```diff
@@ -2,6 +2,50 @@
...
+
+- Type safe `Bound`s [\#462](https://github.com/CosmWasm/cw-plus/issues/462)
+
+Bounds are now type-safe. That means the bound type must match the key / sub-key you are ranging over.
+
+Migration code example:
+
+```diff
+fn list_allowed(
+     limit: Option<u32>,
```
Comment (ethanfrey) [inline MIGRATING.md:17]: Please define type of start_after here

**Review by ethanfrey — COMMENTED** [review]
Nice docs. A few suggestions

File: MIGRATING.md [hunk 2]
```diff
@@ -2,6 +2,50 @@
...
+-        Some(x) => Some(Bound::exclusive(deps.api.addr_validate(&x)?.into_string())),
+-        None => None,
+-    };
++    let addr = maybe_addr(deps.api, start_after)?;
++    let start = addr.as_ref().map(Bound::exclusive);
+
+     let allow = ALLOW_LIST
+         .range(deps.storage, start, None, Order::Ascending)
+```
+Here the `ALLOW_LIST` key is of type `&Addr`. That's why we use `as_ref()` before the `map()` that builds the bound.
```
[the remark this answers is not in the document]
Comment (maurolacy) [reply MIGRATING.md:30]: Will do.

[the remark this answers is not in the document]
Comment (maurolacy) [reply MIGRATING.md:32]: Improved on this to better document `Bound`, and updated some docs examples in the README.

There are still some entries / examples to add, but I think we can merge this and iterate later, before the final 0.12 release.

**Review by ueco-jb — COMMENTED** [review]
Would you mind including this update as well?
https://github.com/CosmWasm/cw-plus/issues/490

**Comment by maurolacy** [comment]
I would, as that is a totally different issue. And, I'm not so acquainted with those changes.

**Review by ueco-jb — APPROVED** [review]
> but I think we can merge this and iterate later, before the final 0.12 release.

Agree.

**Pull request merged**